File: mpi/mpivers.f90
program mpi_vers
! https://github.com/open-mpi/ompi/blob/master/examples/hello_usempif08.f90

use, intrinsic :: iso_fortran_env, only : compiler_version
use mpi_f08, only : MPI_MAX_LIBRARY_VERSION_STRING, mpi_get_library_version, mpi_init, &
  MPI_COMM_SIZE, mpi_comm_world, mpi_finalize, mpi_comm_rank

implicit none (type, external)

integer :: id, Nimg, vlen
character(MPI_MAX_LIBRARY_VERSION_STRING) :: version  ! allocatable not ok


print *,compiler_version()

call MPI_INIT()

call MPI_COMM_RANK(MPI_COMM_WORLD, id)

call MPI_COMM_SIZE(MPI_COMM_WORLD, Nimg)
call MPI_GET_LIBRARY_VERSION(version, vlen)

print '(A,I3,A,I3,A)', 'MPI: Image ', id, ' / ', Nimg, ':', trim(version)

call MPI_FINALIZE()

end program

File: mpi/mpivers.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
   int id, Nimg, ierr, L;
   char version[MPI_MAX_LIBRARY_VERSION_STRING];

   ierr = MPI_Init(&argc,&argv);
   if(ierr != 0) {
      perror("could not init MPI");
      return 1;
   }

   ierr = MPI_Comm_rank(MPI_COMM_WORLD, &id);
   if(ierr != 0) {
      perror("could not get MPI rank");
      return 1;
   }
   MPI_Comm_size(MPI_COMM_WORLD, &Nimg);
   MPI_Get_library_version(version, &L);

   printf("MPI: Image %d / %d : %s",id, Nimg, version);

   ierr = MPI_Finalize();
   if(ierr != 0) {
      perror("could not close MPI");
      return 1;
   }

   return 0;
}
